packages/protocol/contracts/psp/ExampleConditionAndFulfillContract.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+// condition() and fulfill() doesn't have to be in the same contract
+contract ExampleConditionAndFulfillContract {
+    uint256 public constant FOO = 10;
+
+    function condition(bytes calldata data)
+        public
+        view
+        returns (bool checkResult)
+    {
+        // Here, `data` is the API response encoded by the Airnode, i.e., it's
+        // `data` from AirnodeRrp.fulfill(). Let's assume the oracle request returns
+        // one `uint256`.
+        uint256 response = abi.decode(data, (uint256));
+        // Condition checks if the data returned by the API is larger than `FOO` (which means
+        // the Airnode should go ahead with the fulfillment if the API returns a number larger
+        // than `FOO`)
+        checkResult = response > FOO;
+    }
+
+    function fulfill(bytes32 subscriptionId, uint256 timestamp, bytes calldata data)
+        external
+    {
+        // Check if the caller is AirnodePsp
+        // require(msg.sender == address(airnodePsp));
+
+        // Check if you know about the subscription first
+        // require(subscriptionId == ...)
+        // You can also just check if the template ID/parameters is correct
+        // (bytes32 templateId, , , , , bytes parameters) = airnodePsp.subscriptions(subscriptionId);
+        // require(templateId == ...);
+        // require(parameters.length == 0); // No additional parameters
+
+        // Do nothing if the condition doesn't hold.
+        // Although this check is optional, it's highly recommended.
+        require(condition(data), "Condition not met");
```

Comment:
If `condition()` was in a separate contract, this would have been an external call. I'm a bit conflicted about this.

Reply:
> What if API provider is malicious and responds `true`

This call is a static call done to the blockchain provider so they would be the one being malicious (maybe you meant this anyway)

> Airnode would then fulfill the transaction (draining sponsor funds) even though the requester didn't want this data.

Yes, but blockchain providers have other ways of fucking over the Airnode, such as reporting an unnecessarily high gas price

> Shouldn't this be called in the fulfill of AirnodePsp?

The main reason it's not there is that that would have been a second external untrusted low level call, and such calls need to be the last thing you do to avoid a reentrancy attack, but that's not a solution when you have two such calls (and we probably don't want a reentrancy guard just for this). So you can just leave it to the target function to check the condition (and implement the reentrancy guard if `condition()` is not trusted, in this case it is), it will be just as secure.